apps/dashboard/app/controllers/batch_connect/session_contexts_controller.rb

```diff
@@ -74,7 +81,27 @@ def set_render_format
       @render_format = @app.clusters.first.job_config[:adapter] unless @app.clusters.empty?
     end
 
-    # Never trust parameters from the scary internet, only allow the white list through.
+    # Set the rendering format for displaying attributes
+    def set_prefill_templates
+      @prefill_templates ||= begin
+        return {} unless @app.valid?
+
+        json_path = prefill_templates_root.join("*.json")
+        Dir.glob(json_path).map do |path|
+          [File.basename(path, '.json'), File.read(path)]
+        end.to_h
+      end
+    end
+
+    def save_template
+      return unless params[:template].present?
+
+      safe_name = params[:template].gsub(/[\x00\/\\:\*\?\"<>\| ]/, '_')
+      path = prefill_templates_root.join(safe_name.to_s + '.json')
+      path.write(JSON.pretty_generate @session_context.as_json)
```

Comment:
```suggestion
      path.write(@session_context.to_json)
```

Sorry about that! Should be `to_json` not `as_json`. We shouldn't need to `include ActiveModel::Serializers::JSON` because the session_context has the same and it knows how to turn itself to json.

Reply:
> Yeah, but actually I like the idea of pretty_generate (where we would use as_json), since @session_context.to_json will store it as the compressed one, which looks bad if the user tries to edit it manually.

I'm hoping to launch this feature with #2973 implemented so users will edit YAML. Which is to say - by the time this is in users' hands they'll be editing YAML anyway.

Reply:
Reverted as you wish.